### PR TITLE
feat: implement region migration manager

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Implement the region migration manager.

More tests of managers will be introduced with integration tests(p.s. Previously, unit tests have covered region migration procedure flow-control tests)

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
#2700

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]

File: src/meta-srv/src/error.rs [hunk 1]
```diff
@@ -32,6 +32,12 @@ use crate::pubsub::Message;
 #[snafu(visibility(pub))]
 #[stack_trace_debug]
 pub enum Error {
+    #[snafu(display("Another migration procedure is running for region: {}", region_id,))]
```
Comment (MichaelScofield) [inline src/meta-srv/src/error.rs:35]:
```suggestion
    #[snafu(display("Another migration procedure is running for region: {}", region_id))]
```

File: src/meta-srv/src/procedure/region_migration/manager.rs [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+            .as_ref()
+            .context(error::UnexpectedSnafu {
+                violated: "Region route leader peer is not found",
+            })?;
+
+        if leader_peer.id == task.to_peer.id {
+            return Ok(true);
+        }
+
+        Ok(false)
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/region_migration/manager.rs:201]:
```suggestion
        Ok(leader_peer.id == task.to_peer.id)
```

File: src/meta-srv/src/procedure/region_migration/manager.rs [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+    }
+
+    /// Throws an error if `leader_peer` is not the `from_peer`.
+    async fn verify_region_leader_peer(
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/region_migration/manager.rs:205]: No need to async.

**Comment by WenyXu** [comment]
@MichaelScofield PTAL

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**